public/locations-list-map.js

```diff
@@ -316,10 +367,186 @@ function initMap(stateFilter) {
   map = new google.maps.Map(element);
 
   showMarkers(data_by_location, { states: stateFilter }, !stateFilter);
+
+  // TODO (patricknelson): This showMarkers call above has to be duplicated due to crossing of concerns of this function
+  //   (init map vs. initial map state). See resetMap() function. Also, ideally the initialization of filters would happen
+  //  OUTSIDE of this method. Also, to reduce merge conflicts, the initialization of the "initial_marker_filters" is
+  //  defined here for now, but ideally it would instead be defined in jQuery DOM ready function instead.
+  initial_marker_filters = { states: stateFilter };
+
+  // Initialize autosuggest/search field above the map.
+  initMapSearch();
 }
 
+
+
+/**********************************
+ * BEGIN MAP SEARCH FUNCTIONALITY *
+ **********************************/
+
+/**
+ * Responsible for initializing the search field and links below the search field (e.g. use location, reset map, etc).
+ */
+function initMapSearch() {
+  // If disabled, hide the search fields and don't bother attaching any functionality to them.
+  if (!showMapSearch) {
+    $('.map-search-wrap').hide();
+    return;
+  }
+
+  // Initialize the map search autocompleter.
+  autocomplete = new google.maps.places.Autocomplete(
+    document.getElementById('map-search'), {types: ['geocode']}
+  );
+
+  // Avoid paying for data that you don't need by restricting the set of
+  // place fields that are returned to just the address components.
+  autocomplete.setFields(['geometry']);
+
+  // When the user selects an address from the drop-down, populate the
+  // address fields in the form.
+  autocomplete.addListener('place_changed', loadSearchInMap);
+
+  // Setup event listeners for map action links.
+  $('#use-location').on('click', (e) => {
+    e.preventDefault();
+    centerMapToMarkersNearUser();
+  });
+
+  $('#reset-map').on('click', (e) => {
+    e.preventDefault();
+    resetMap();
+  });
+}
+
+
+
+/**
+ * Strictly responsible for resetting the map to it's initial state on page load WITHOUT user's location (since we have
+ * a link to link to go back to that appearance).
+ */
+function resetMap() {
+  // TODO THIS SHOULD HAVE TWO FUNCTION CALLS INSTEAD OF ONE:
+  //  1.) Show initial markers based on initial filters (e.g. which state[s]?)
+  //  2.) Reposition map (e.g. initial state?)
+  //  However, repositioning is ALSO happening deep inside showMarkers() function which has made itself responsible for
+  //  too many tasks.
+  showMarkers(window.data_by_location, initial_marker_filters, !initial_marker_filters);
```

Comment:
Then there's the question of what happens when someone loads the page, clicks a filter, does a location search, then resets the map.

Reply:
Yep... there needs to be better separation of concerns, but we'll move this into beta meaning it'll be disabled by default. @drwestco please note your concerns for later and we should address them once the PR is merged as this will be disabled when we merge into `master` (working on that right now)

Reply:
my fix for this right now, due to that tight binding until my recommendations are adopted would be something along the lines of:

```js
  let showNearest = false;
  if (!initial_marker_filters.states) showNearest = !initial_marker_filters.states; // ... just in case "states" property goes missing...
  showMarkers(window.data_by_location, initial_marker_filters, showNearest);
```